### PR TITLE
[smart_holder] git merge master

### PR DESCRIPTION
**Pull request by rwgk**
## Description
Helper/scratch PR for testing.

## Suggested changelog entry:

```rst

```

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by rwgk** [comment]
Force push was after
* git rebase smart_holder, after #3807 was merged there
* 2nd git merge master, after #3811 was merged there
The end result is exactly identical to what was globally tested last night.
Waiting for the CI here to finish, out of an abundance of caution, JIC I messed up with the git commands.

**Comment by rwgk** [comment]
The one CI failure is an infrastructure flake (pypy-3.7 • windows-2022 • x64). Safe to ignore, based on all the previous testing.

**Pull request merged**
